src/Motors/MotorModule.c

```diff
@@ -0,0 +1,262 @@
+#include "MotorModule.h"
+#include "/time/timeInterrupt.h"
+#include <avr/io.h>
+#include <avr/sfr_defs.h>
+#include <avr/delay.h>
+#include <stdbool.h>
+
+/* ========================= Setup ================================= */
+#define blinkInterval 400
+uint32_t currentIndicater = 0;
+uint32_t previousIndicater = 0;
+
+// debug will allow the run of tests
+void initMotorModule(bool test) {
+	// Control Pins
+	DDRB |= 1<<PINB0 | 1<<PINB3 | 1<<PINB1 | 1<<PINB2;	// IN1/2 | IN3/4 | ENA | ENB
+	
+	// Timer/Counter0
+	TCCR1A |= 1<<WGM11 | 1<<WGM10 | 1<<COM1A1 | 1<<COM1B1;
+	TCCR1B |= 1<<CS12 | 1<<CS10;
+	
+	// Indicator LED lights
+	DDRC |= (1<<DDC1) | (1<<DDC2);
+	
+	// Debug & testing
+	if(test){
+		setAllMotorSpeed(240);
+		setMotorDirection(true);
+		setMotorDirection(false);
+		steerRightSimple(200);
+		steerLeftSimple(200);
+		zeroDegreeLeft(200);
+		zeroDegreeRight(200);
+		setAllMotorSpeed(250);
+		// Frieda: The time delays are not really working.
+		//				Maybe need some replacement with interrupt timer
+		//				But the functions can be tested in the debugger with breakpoints.
+		//				The functions to call are prepared above
+/*		motorLinearIncreaseTest();
+		motorHighDrasticSpeedChange();
+		forwardBackwardsDrivingTest();
+		simpleSteeringTest();
+		complexSteeringTest();
+		stopMotors();
+		*/
+	}
+}
+
+/* ========================= indicator functions ================================= */
+
+void indicatorLeft(){
+	currentIndicater = millis();
```

Comment:
I thought millis(); was no longer a thing in microchip studio is this a library you're using?
ps. i want to know for my own sake not per se anything wrong with this

Reply:
As you know in this period we have to use interrupts and timer/counter for clocks and stuff. Because we also want to debounce buttons or time specific instructions we need a form to control the time.
For this I added our own interrupt timer where we can measure Millis. The files are in the folder timer, there is also an initMillis that is called in the main.c.
If you want I can explain it tomorrow in more detail if wanted or look up week 4 in the MIC study guide.